### PR TITLE
[Form] Remove deprecated code

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       |  -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by yceruto** [comment]
I will continue with the rest tomorrow if no one takes care of it before. Cheers!

**Review by derrabus — APPROVED** [review]
Please apply the CS fix suggested by Fabbot. Other than that, I'm happy with the changes. 👍🏻

**Comment by yceruto** [comment]
I fixed the issue spotted by fabbot.io (the last one is a false positive)

**Comment by nicolas-grekas** [comment]
(rebase needed)

**Comment by yceruto** [comment]
(rebased)

**Review by nicolas-grekas — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Form/FormConfigBuilderInterface.php [hunk 1]
```diff
@@ -251,4 +249,12 @@ public function setAutoInitialize(bool $initialize);
...
      */
     public function getFormConfig();
+
+    /**
+     * Sets the callback that will be called to determine if the model
+     * data of the form is empty or not.
+     *
+     * @return $this The configuration object
+     */
+    public function setIsEmptyCallback(?callable $isEmptyCallback);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/FormConfigBuilderInterface.php]: let's add `static` as return type

**Review by nicolas-grekas — COMMENTED** [review]
Can you please check psalm reports and let us know if we they're false positive or not?
Please rebase also

File: src/Symfony/Component/Form/FormConfigBuilderInterface.php (hunk 1 above)
Comment (yceruto) [reply src/Symfony/Component/Form/FormConfigBuilderInterface.php]: Done.

**Comment by yceruto** [comment]
> Can you please check psalm reports and let us know if we they're false positive or not?

It looks like a false positive for me.

(rebased)

File: src/Symfony/Component/Form/FormConfigBuilderInterface.php [hunk 2]
```diff
@@ -251,4 +249,12 @@ public function setAutoInitialize(bool $initialize);
      * @return FormConfigInterface
      */
     public function getFormConfig();
+
+    /**
+     * Sets the callback that will be called to determine if the model
+     * data of the form is empty or not.
+     *
+     * @return $this The configuration object
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/FormConfigBuilderInterface.php]: let's remove the end and keep only `@return $this`

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @yceruto.

**Pull request merged**